test/cache.py

```diff
@@ -0,0 +1,138 @@
+logging.basicConfig(level=logging.DEBUG)
+
+
+import json
+
+
+
+class CacheTest(TestCase):
+    def setUp(self):
+        """Perpare to run test"""
+
+        super(CacheTest, self).setUp()
+
+        self.tmpdir = tempfile.mkdtemp()
+        self.topdir = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+        # monkeypatch
+        cache.LMOD_CONFIG = './test/data/lmodrc.lua'
+
+    def tearDown(self):
+        """Clean up after running test"""
+        shutil.rmtree(self.tmpdir)
+
+        super(CacheTest, self).tearDown()
+
+    def test_lmod_conf(self):
+        self.assertEqual(get_lmod_conf(),
+                         {'timestamp': '/apps/gent/lmodcache/timestamp', 'dir': './test/data'})
+
+    def test_read_write_json(self):
+        """Test reading and writing the json data"""
+        self.assertEqual(get_json_filename(), './test/data/modulemap.json')
+
+        jsonfilename = os.path.join(self.tmpdir, 'some.json')
+        clustermap = {'a': 'b'}
+        softmap = {'g': 'h'}
+        write_json(clustermap, softmap, filename=jsonfilename)
+        self.assertEqual(json.loads(open(jsonfilename).read()), {'clusters': clustermap, 'software': softmap})
+        self.assertEqual(read_json(filename=jsonfilename), (clustermap, softmap))
+
+    def test_make_json(self):
+        """More or less the code from convert_lmod_cache_to_json"""
+        cachefile = os.path.join(get_lmod_conf()['dir'], CACHEFILENAME)
+        self.assertEqual(cachefile, './test/data/spiderT.lua')
+
+        mpathMapT, spiderT = get_lmod_cache(cachefile)
+
+        mpath = '/apps/gent/CO7/cascadelake-volta-ib-PILOT/modules/all'
+        name = 'Autoconf'
+        module = 'Autoconf/2.69-GCCcore-8.2.0'
+        version = '2.69-GCCcore-8.2.0'
+
+        # only test one value
+        self.assertEqual(mpathMapT[mpath], {'cluster/.joltik': '/etc/modulefiles/vsc'})
+        # only test one value
+        self.assertEqual(spiderT[mpath][name]['fileT'][module]['Version'], version)
+
+        clustermap, mpmap = cluster_map(mpathMapT)
+        # test 2 values, one hidden cluster
+        self.assertEqual(clustermap['banette'], 'cluster/.banette')
+        self.assertEqual(clustermap['delcatty'], 'cluster/delcatty')
+
+        # test one, multivalue
+        self.assertEqual(mpmap['/apps/gent/CO7/haswell-ib/modules/all'], ['golett', 'swalot', 'phanpy'])
+
+        # only two mpaths in the mocked spiderT
+        mpath2 = '/apps/gent/CO7/skylake-ib/modules/all'
+        newmpmap = {mpath: mpmap[mpath], mpath2: mpmap[mpath2]}
+        softmap = software_map(spiderT, newmpmap)
+        self.assertEqual(softmap['Autotools'],
+                         {'.default': {'joltik': '20180311-GCCcore-8.3.0'},
+                          '20180311-GCCcore-8.2.0': ['joltik'],
+                          '20180311-GCCcore-8.3.0': ['joltik']})
+        self.assertEqual(softmap['Bazel'],
+                         {'.default': {'skitty': '0.20.0-GCCcore-8.2.0',
```

Comment:
Maybe add a comment to clarify that this default was set manually in the `spiderT.lua` used for the tests?

Since setting an old version to default is quite strange :)

Reply:
No comment?

Reply:
done